src/main/osd/osd_elements.c

```diff
@@ -240,6 +240,44 @@ static void osdFormatCoordinate(char *buff, char sym, int32_t val)
 }
 #endif // USE_GPS
 
+void osdFormatDistanceString(char *result, int distance, char leadingSymbol)
+{
+    const int convertedDistance = osdGetMetersToSelectedUnit(distance);
+    char unitSymbol;
+    char unitSymbolExtended;
+    int unitTransition;
+    char buff[10];
+
+    result[0] = leadingSymbol;
+    if (leadingSymbol != SYM_NONE) {
```

Comment:
To me the use of `SYM_NONE` in this way is quite confusing - in MAX7456 terminology 0x00 is 'whitespace' (or at least the character that is used to clear the screen on initialisation). It took me a while to figure out that you are using it here not to place a whitespace, but to place a C string terminator.

Reply:
Well I didn't add `SYM_NONE` to act as a string terminator. The point was to support the case when no leading symbol was wanted in the formatted string - in other words "symbol none". There are multiple places where `osdFormatDistanceString()` is used and some (the elements) want a leading symbol and others (the stats) don't use a leading symbol. So `SYM_NONE` is just used to be the case where no leading symbol is desired. The fact that it was 0x00 and could be used as a string terminator for an empty string was just a side benefit I took advantage of to simplify the code. So the code is unconditionally adding the symbol to the first character, and conditionally adding a `0x00` terminator if the symbol is not `SYM_NONE` (since it's already a terminator). That's all that's going on.

And actually in the MAX7456 driver we use 0x20 as the screen clearing character in the buffer copies - not 0x00. The hardware clear would set the internal MAX7456 buffer to 0x00 but we never use that because of the need to keep a shadow copy of the hardware buffer and do the incremental (only characters that changed) update logic (we don't want white space to be a "change"). But that's a whole separate discussion.

Reply:
> The fact that it was 0x00 and could be used as a string terminator for an empty string was just a side benefit I took advantage of to simplify the code.

I personally do not think that the gain in simplicity outweighs the fact that it makes it hard to understand in this case. I think @ledvinap's suggestion is a better alternative.